extensions/amp-story-panning-media/0.1/test/test-amp-story-panning-media.js

```diff
@@ -28,22 +28,48 @@ describes.realWin(
   (env) => {
     let win;
     let element;
+    let panningMedia;
 
-    beforeEach(() => {
-      win = env.win;
+    function appendAmpImg(parent, path) {
+      const ampImg = createElementWithAttributes(win.document, 'amp-img', {
+        'src': path,
+        'width': '4000',
+        'height': '3059',
+      });
+      parent.appendChild(ampImg);
+    }
+
+    async function createAmpStoryPanningMedia(imagePath) {
+      const pageEl = win.document.createElement('amp-story-page');
+      pageEl.id = 'page1';
       element = createElementWithAttributes(
         win.document,
         'amp-story-panning-media',
         {
-          layout: 'fill',
+          'layout': 'fill',
+          'x': '50%',
+          'y': '50%',
+          'zoom': '1',
         }
       );
-      win.document.body.appendChild(element);
+      if (imagePath) {
+        appendAmpImg(element, imagePath);
+      }
+      pageEl.appendChild(element);
+      win.document.body.appendChild(pageEl);
+
+      panningMedia = await element.getImpl();
+    }
+
+    beforeEach(() => {
+      win = env.win;
     });
 
-    it('should contain "hello world" when built', async () => {
-      await element.whenBuilt();
-      expect(element.querySelector('div').textContent).to.equal('hello world');
+    it('should build', async () => {
+      await createAmpStoryPanningMedia('/examples/img/conservatory-coords.jpg');
+      expect(() => {
+        panningMedia.layoutCallback();
```

Comment:
This being asynchronous, we should add a "await" to check that most of that method runs before we exit the test. I know you'll have to work with the `amp-img` loading, but Tom and I wrote some tests you can find inspiration from in the 360 extension.

Reply:
Sounds good. Added tests to:

- throw if nested amp-img is missing
- set transform of image element from attributes

I think that covers the core concepts implemented in this PR.

Reply:
Did you check that the test waits for the layoutCallback to finish before ending? I think you need to block it so that it waits on layoutCallback. 

e.g. using return:

```javascript
    it('should build', async () => {
      await createAmpStoryPanningMedia('/examples/img/conservatory-coords.jpg');
      return expect(() =>  panningMedia.layoutCallback()).to.not.throw();
    }
```